src/FreeDSx/Snmp/Protocol/ClientProtocolHandler.php

```diff
@@ -270,7 +264,12 @@ protected function generateMessageHeader(RequestInterface $request, array $optio
     {
         $header = new MessageHeader($this->generateId(0));
 
-        if ($options['use_auth'] || $options['use_priv']) {
+        $useAuth = $options['use_auth'];
+        $usePriv = $options['use_priv'];
+        assert(is_bool($useAuth));
+        assert(is_bool($usePriv));
```

Comment:
Same as the other assert comment I guess. Could catch this just once in the client.